ndscan/fitting/fitting.py

```diff
@@ -0,0 +1,269 @@
+import numpy as np
+import functools
+import collections
+from typing import Dict, List, Optional, Tuple, Union
+from scipy import optimize
+
+
+# TODO: type annotations! (What's the best way of annotating np arrays?)
```

Comment:
This is something I find annoying too -- I'd like to be able to say what dimensions I expect the array to be.

I will send you a link to how I've handled this in an OxIonics project, but afaik there's nothing built into numpy for this. You'd just have to use `np.ndarray`.

Reply:
I've had a look at this as well in the context of NAC3, and came to the same conclusion; this isn't nicely supported in NumPy at all. There is some support for type annotations now (e.g. you can do `ndarray[int32, 2]`), but IIRC it isn't even clearly defined what the index types should even be.